### PR TITLE
chore: spread cleanup

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox run -m lint`?
- [ ] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -276,36 +276,34 @@ suites:
...
+     else
+       snap install core24 --edge
+     fi
 
- tests/spread/core24/environment/:
+
+ tests/spread/core24-suites/environment/:
    summary: core24 environment tests
    systems:
      - ubuntu-22.04*
```
Comment (tigarmo) [inline spread.yaml:294]: A bunch of these tests run in destructive mode, are they supposed to run on jammy still?

File: spread.yaml [hunk 2]
```diff
@@ -276,36 +276,34 @@ suites:
    summary: core24 specific tests
    systems:
      - ubuntu-24.04*
+   environment:
+     SNAPCRAFT_BUILD_ENVIRONMENT: ""
+   prepare: |
+     # Workaround to be able to test the snaps
+     if snap install core24; then
```
Comment (lengau) [inline spread.yaml:283]: I take it we won't have a point where it's released to `beta` or `candidate` that we want to check?

File: tests/spread/core24/linters-file/task.yaml [hunk 3]
```diff
@@ -1,13 +1,17 @@
 summary: Run snapcraft lint on a snap file.
 
+environment:
+  SNAP: lint-file
+
 restore: |
-  rm -f ./*.snap ./*.assert
+  cd "${SNAP}"
+  rm -f ./*.snap ./*.assert ./linter_output.txt
+  rm -f ./*.snap
```
Comment (lengau) [inline tests/spread/core24/linters-file/task.yaml:9]:
```suggestion
  rm -f ./*.snap ./*.assert ./linter_output.txt
```

File: spread.yaml (hunk 1 above)
Comment (sergiusens) [reply spread.yaml:294]: good catch! I thought I did that update

File: tests/spread/core24/linters-pack/task.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+environment:
+  SNAP/classic: classic
+  SNAP/classic_libc: classic-libc
+  SNAP/library_gnome: library-gnome
+  SNAP/library_ignore_missing: library-ignore-missing
+  SNAP/library_ignore_unused: library-ignore-unused
+  SNAP/library_ignored_mixed: library-ignored-mixed
+  SNAP/library_missing: library-missing
+  SNAP/library_unused: library-unused
```
Comment (lengau) [inline tests/spread/core24/linters-pack/task.yaml:11]: Nice use of variants here!

File: tests/spread/core24/try/task.yaml
Comment (lengau) [inline tests/spread/core24/try/task.yaml:1]: What replaces this test? (Or why don't we need it?)

**Review by lengau — COMMENTED** [review]
I like this - just a few questions.

File: tests/spread/core24/try/task.yaml
Comment (sergiusens) [reply tests/spread/core24/try/task.yaml:1]: we said we were not doing try for core24 and do something different (use the project root directly)

File: spread.yaml (hunk 2 above)
Comment (sergiusens) [reply spread.yaml:283]: this as is does no harm, but it is a workaround to have the base as snapd does not install it by default, by the time we reach beta or candidate core24 will look pretty similar across the channel risks (security updates mostly)

File: tests/spread/core24/linters-pack/task.yaml (hunk 4 above)
Comment (sergiusens) [reply tests/spread/core24/linters-pack/task.yaml:11]: thanks!

File: tests/spread/core24/linters-file/task.yaml (hunk 3 above)
Comment (sergiusens) [reply tests/spread/core24/linters-file/task.yaml:9]: I will do this when I rebase the PR you took over

File: spread.yaml (hunk 1 above)
Comment (sergiusens) [reply spread.yaml]: ok, with the current setup:

```
2024-03-05T14:38:09.8946667Z     - google:ubuntu-22.04-64:tests/spread/core24-suites/environment/test-variables:test_variables
2024-03-05T14:38:09.8949593Z     - google:ubuntu-22.04-64:tests/spread/core24-suites/manifest/manifest-creation:envvars
2024-03-05T14:38:09.8953244Z     - google:ubuntu-22.04-64:tests/spread/core24-suites/patchelf/classic-patchelf
2024-03-05T14:38:09.8970397Z     - google:ubuntu-22.04-64:tests/spread/core24-suites/patchelf/strict-patchelf
```

With Ubuntu 24.04 -> https://github.com/canonical/snapcraft/pull/4634/commits/39541c66148d30fc9a8f8d9bb6641639267f1e90

```
    - google:ubuntu-24.04-64:tests/spread/core24-suites/environment/paths:staged_common_library
    - google:ubuntu-24.04-64:tests/spread/core24-suites/environment/test-variables:test_variables
    - google:ubuntu-24.04-64:tests/spread/core24-suites/manifest/manifest-creation:envvars
    - google:ubuntu-24.04-64:tests/spread/core24-suites/patchelf/classic-patchelf
    - google:ubuntu-24.04-64:tests/spread/core24-suites/patchelf/strict-patchelf
    - google:ubuntu-24.04-64:tests/spread/core24-suites/scriptlets/scriptlet-failures
```

The last one is probably my old version of spread, the patchelf ones I think I can fix up to the point where they properly fail again due to our lack of classic confinement support -> https://github.com/canonical/snapcraft/pull/4634/commits/e8f688eb0e9a3061833e39d306f53fd96d33ea7c

File: tests/spread/core24/try/task.yaml
Comment (lengau) [reply tests/spread/core24/try/task.yaml:1]: Got it, thanks.

**Review by syu-w — APPROVED** [review]

**Review by tigarmo — APPROVED** [review]

**Pull request merged**